PhoneBook.Kakurokan/Program.cs

```diff
@@ -0,0 +1,185 @@
+        .PageSize(15)
+        .AddChoices(new[] {
+            "Yes", "No"
+        })); ;
+
+            if (choice == "Yes")
+            {
+                dataAccess.Remove(dataAccess.Contact.Single(a => a.Id == id));
+                dataAccess.SaveChanges();
+            }
+            DisplayReturningTomenu();
+
+        }
+
+        public static void UpdateContact(DataAccess dataAccess)
+        {
+            int id = SelectContact(dataAccess);
+
+            ContactModel contact = CreateContact();
+
+            var oldContact = dataAccess.Contact.First(a => a.Id == id);
+            oldContact.Name = contact.Name;
+            oldContact.PhoneNumber = contact.PhoneNumber;
+            oldContact.Email = contact.Email;
+            dataAccess.SaveChanges();
+            DisplayReturningTomenu();
+
+        }
+
+        public static int SelectContact(DataAccess dataAcess)
+        {
+            var option =
+    new SelectionPrompt<string>()
+        .Title("Select a contact to [Red]delete[/]?")
+        .PageSize(10)
+        .MoreChoicesText("[grey](Move up and down to reveal more contacts)[/]");
+
+            foreach (var contact in dataAcess.Contact)
+            {
+                option.AddChoice("Id: " + contact.Id + "; Name: " + contact.Name + "; Email: " + contact.Email + "; Phonenumber: " + contact.PhoneNumber);
+            }
+
+            string selectedContact = AnsiConsole.Prompt(option);
+
+            int id = int.Parse(Regex.Match(selectedContact, @"Id:\s*([0-9]+)").Value.Remove(0, 3));
+
+            return id;
+        }
+
+        public static ContactModel CreateContact()
+        {
+            string name = AnsiConsole.Ask<string>("What´s the [Red]name[/] of your contact?");
+
+            string email = "";
+
+            while (email == "")
+            {
+                email = AnsiConsole.Ask<string>("What´s the [Red]email[/] of your contact?");
+                if (!MailAddress.TryCreate(email, out var mailAddress))
+                {
+                    email = "";
+                    AnsiConsole.Markup("[Red]Invalid email![/]");
+                    AnsiConsole.WriteLine();
+                }
+            }
+
+            string number = "";
+
+            while (number == "")
+            {
+                number = AnsiConsole.Ask<string>("What´s the [Red]phonenumber[/] of your contact? <(555) 444-1234>");
+                if (!Regex.IsMatch(number, @"^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$"))
+                {
+                    number = "";
+                    AnsiConsole.Markup("[Red]Invalid number![/]");
+                    AnsiConsole.WriteLine();
+                }
+            }
+
+            return new(name, email, number);
+        }
+
+        public static void DisplayReturningTomenu()
+        {
+            AnsiConsole.Clear();
+            AnsiConsole.Status().Start("Returning to menu...", ctx =>
+            {
+                Thread.Sleep(2000);
```

Comment:
Haha is this an exercise in Zen or mindfulness 🤣 . Nice touch with the loading screen, but it took for ever 😆